pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -355,27 +420,27 @@ def set_distributed_mode(self):
         # special case with TPUs
         elif self.distributed_backend == 'tpu':
             self._device_type = DeviceType.TPU
-        # set all other requested distrib. types adn if it was not set in the
+        # set all other requested distrib. types and if it was not set in the
         elif self.distributed_backend and self._distrib_type is None:
             self._distrib_type = DistributedType(self.distributed_backend)
 
         # unless you request explicitly for CPU and some GPU are available use them
         _on_cpu = self.distributed_backend and 'cpu' in self.distributed_backend
-        if (self.num_gpus > 0 and not _on_cpu):
+        if self.num_gpus > 0 and not _on_cpu:
             self._device_type = DeviceType.GPU
 
-        _distrib_types = (DistributedType.DP, DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2)
+        # _distrib_types = (DistributedType.DP, DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2)
```

Comment:
@awaelchli can we remove this?

Reply:
I uncommented it, I wasn't sure but the condition doesn't make sense to me and it caused ddp_cpu to fail selecting the right flags for the plugin selection. I think it has to be removed yes

Reply:
re-added, it was indeed necessary.